libs/community/langchain_community/document_loaders/blob_loaders/cloud_blob_loader.py

```diff
@@ -0,0 +1,294 @@
+                           This forces an iteration through all matching files
+                           to count them prior to loading them.
+
+        Examples:
+
+            .. code-block:: python
+                from langchain_community.document_loaders.blob_loaders import CloudBlobLoader
+
+                # Load a single file.
+                loader = CloudBlobLoader("s3://mybucket/id") # az://
+
+                # Recursively load all text files in a directory.
+                loader = CloudBlobLoader("az://mybucket/id", glob="**/*.txt")
+
+                # Recursively load all non-hidden files in a directory.
+                loader = CloudBlobLoader("gs://mybucket/id", glob="**/[!.]*")
+
+                # Load all files in a directory without recursion.
+                loader = CloudBlobLoader("s3://mybucket/id", glob="*")
+
+                # Recursively load all files in a directory, except for py or pyc files.
+                loader = CloudBlobLoader(
+                    "s3://mybucket/id",
+                    glob="**/*.txt",
+                    exclude=["**/*.py", "**/*.pyc"]
+                )
+        """  # noqa: E501
+        from cloudpathlib import AnyPath
+
+        url_parsed = urlparse(str(url))
+
+        if url_parsed.scheme == "file":
+            url = url_parsed.path
+
+        if isinstance(url, str):
+            self.path = AnyPath(url)
+        else:
+            self.path = url
+
+        self.glob = glob
+        self.suffixes = set(suffixes or [])
+        self.show_progress = show_progress
+        self.exclude = exclude
+
+    def yield_blobs(
+        self,
+    ) -> Iterable[Blob]:
+        """Yield blobs that match the requested pattern."""
+        iterator = _make_iterator(
+            length_func=self.count_matching_files, show_progress=self.show_progress
+        )
+
+        for path in iterator(self._yield_paths()):
+            # yield Blob.from_path(path)
+            yield self.from_cloud_path(path)
+
+    def _yield_paths(self) -> Iterable["AnyPath"]:
+        """Yield paths that match the requested pattern."""
+        if self.path.is_file():  # type: ignore
+            yield self.path
+            return
+
+        paths = self.path.glob(self.glob)  # type: ignore
+        for path in paths:
+            if self.exclude:
+                if any(path.match(glob) for glob in self.exclude):
+                    continue  # FIXME
+            if path.is_file():
+                if self.suffixes and path.suffix not in self.suffixes:
+                    continue  # FIXME
+                yield path
+
+    def count_matching_files(self) -> int:
+        """Count files that match the pattern without loading them."""
+        # Carry out a full iteration to count the files without
+        # materializing anything expensive in memory.
+        num = 0
+        for _ in self._yield_paths():
+            num += 1
+        return num
+
+    @classmethod
+    def from_cloud_path(
```

Comment:
* Could you replace with static method? 
* Could you rename to `from_path`? (Since it looks like file system is in fact supported, and document which kinds of paths are supported)
* I am assuming this has been attached as a method on the class to avoid exposing the overridden CloudBlob to end users?

Reply:
done